Sources/BSON/Integers+BSONValue.swift

```diff
@@ -1,6 +1,66 @@
 import NIO
 
 extension Int32: BSONValue {
+    /*
+    * Initializes an `Int32` from ExtendedJSON
+    *
+    * Parameters:
+    *   - `json`: a `JSON` representing the canonical or relaxed form of ExtendedJSON for an `Int32`.
+    *   - `keyPath`: an array of `String`s containing the enclosing JSON keys of the current json being passed in.
+    *              This is used for error messages.
+    *
+    * Returns:
+    *   - `nil` if the provided value is not an `Int32`.
+    *
+    * Throws:
+    *   - `DecodingError` if `json` is a partial match or is malformed.
+    */
+    internal init?(fromExtJSON json: JSON, keyPath: [String]? = nil) throws {
```

Comment:
I think we always want a keyPath provided, since this will only be used in the context of decoding things. So I think we want this to just be a `[String]` with no default value.

Reply:
Now that I think about this some more, we may not always have a key path, since we'll be supporting decoding non-array/object stuff at the top level. So I think you're right that this should be optional, though I still think we can remove this default value so that we don't forget to pass it through as necessary.

Reply:
I think I've changed my mind again and I promise this will be the last: I think we actually don't want it to be optional, but for top level stuff just have an empty array `[]`. In the error we can just omit the prefix if the array is empty.

Reply:
if we remove the `=nil` do we have to manually pass in `nil` each time?